src/main/java/com/greenfoxacademy/goddesstribesbackend/controllers/BuildingController.java

```diff
@@ -71,11 +71,15 @@ public BuildingController(KingdomService kingdomService, ProductionService produ
   @ApiImplicitParams({@ApiImplicitParam(name = "token", value = "Authorization token", required = true, dataType = "string", paramType = "header")})
   @ApiResponses(value = {@ApiResponse(code = 200, message = "OK", response = BuildingDTO.class), @ApiResponse(code = 409, message = "Id not found", response = ErrorMessage.class)})
   @GetMapping("/kingdom/buildings/{id}")
-  public ResponseEntity<Object> mockRenderBuilding(@PathVariable Long id) {
-    if (MockData.farm.getId() == id) {
-      return ResponseEntity.status(200).body(MockData.farm);
+  public ResponseEntity<Object> findBuilding(@PathVariable(name = "id") Long buidingId) {
+    String username = SecurityContextHolder.getContext().getAuthentication().getName();
+    Kingdom kingdom = kingdomService.findKingdomByUsername(username);
+    Building building = buildingService.findBuildingByKingdomAndBuildingId(kingdom.getId(), buidingId);
+
+    if (building == null) {
+      return ResponseEntity.status(404).body(new ErrorMessage("Id not found"));
```

Comment:
az Id megvan, a Building azzal az Id-vel nincs meg..

Reply:
Úgy érted, hogy írjam át a hibaüzenetet, mert ez így hülyeség? (Így van benne a swagger api-ban...)
Ez ok? "The building with this id not found"

Reply:
Ok, I changed the error message.